### PR TITLE
nixos/nscd: add option to use nsncd, init nsncd

### PR DESCRIPTION
**Pull request by flokli**
###### Description of changes
This provides a `services.nscd.enableNsncd`, which will use `nsncd` instead of `nscd` to provide NSS lookups.

We use [nix-community/nsncd](https://github.com/nix-community/nsncd/tree/host-lookups-and-noops) due to https://github.com/twosigma/nsncd/discussions/44:

> We'd definitely like to collaborate with you and work to incorporate your changes in our project. But this is not the main day-to-day job for any of us, so if we're too slow to respond feel free to keep experimenting in your forks and we'll try to keep up with your pace.

The nscd test has been included, to ensure it works the way it's supposed to.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/system/nscd.nix [hunk 1]
```diff
@@ -89,14 +104,14 @@ in
 
         environment = { LD_LIBRARY_PATH = nssModulesPath; };
 
-        restartTriggers = [
+        restartTriggers = lib.optionals cfg.enableNsncd ([
```
Comment (xaverdh) [inline nixos/modules/services/system/nscd.nix:107]: should this condition be `!cfg.enableNsncd`?

Comment (flokli) [reply nixos/modules/services/system/nscd.nix]: Yes, you're right. Updated, thanks!

**Review by ajs124 — APPROVED** [review]

File: nixos/tests/nscd.nix [hunk 2]
```diff
@@ -21,10 +21,31 @@ in
       192.0.2.1 somehost.test
     '';
 
+    systemd.services.sockdump = {
```
Comment (xaverdh) [inline nixos/tests/nscd.nix:24]: maybe `sockdump-nscd`, since this runs it on the `nscd` socket?

**Review by xaverdh — APPROVED** [review]

**Comment by rnhmjoj** [comment]
Very nice. Now that we have this, what about disabling caching by default?
I think the issues caused by cached DNS queries are surprising enough that running nscd should be a conscious decision.

**Comment by flokli** [comment]
Flipping the default is something I'd like to do for the next release (after 22.11), but it needs some more changelog and documentation. See https://github.com/NixOS/nixpkgs/pull/197986.

**Pull request merged**